File: Practica 2 - Java RMI/GestorBibliotecaService/src/gestor/biblioteca/service/models/TDatosRepositorio.java
package gestor.biblioteca.service.models;

import java.io.Serializable;
import java.util.Objects;

/**
 *
 * @author Saúl Rodríguez Naranjo
 */
public class TDatosRepositorio implements Serializable
{
    private String repositoryName;
    private String repositoryAddress;
    private int numberOfBooks;
    private String repositoryFilePath;
    
    private BookRepository bookRepository;

    public TDatosRepositorio()
    {
    }

    public TDatosRepositorio(String repositoryName, String repositoryAddress, int numberOfBooks,
            BookRepository bookRepository, String repositoryFilePath)
    {
        this.repositoryName = repositoryName;
        this.repositoryAddress = repositoryAddress;
        this.numberOfBooks = numberOfBooks;
        this.bookRepository = bookRepository;
        this.repositoryFilePath = repositoryFilePath;
    }

    public String getRepositoryName()
    {
        return repositoryName;
    }

    public void setRepositoryName(String repositoryName)
    {
        this.repositoryName = repositoryName;
    }

    public String getRepositoryAddress()
    {
        return repositoryAddress;
    }

    public void setRepositoryAddress(String repositoryAddress)
    {
        this.repositoryAddress = repositoryAddress;
    }

    public int getNumberOfBooks()
    {
        return numberOfBooks;
    }

    public void setNumberOfBooks(int numberOfBooks)
    {
        this.numberOfBooks = numberOfBooks;
    }

    public BookRepository getBookRepository()
    {
        return bookRepository;
    }

    public void setBookRepository(BookRepository bookRepository)
    {
        this.bookRepository = bookRepository;
    }

    public String getRepositoryFilePath()
    {
        return repositoryFilePath;
    }

    public void setRepositoryFilePath(String repositoryFilePath)
    {
        this.repositoryFilePath = repositoryFilePath;
    }

    @Override
    public int hashCode()
    {
        int hash = 5;
        hash = 89 * hash + Objects.hashCode(this.repositoryName);
        hash = 89 * hash + Objects.hashCode(this.repositoryAddress);
        hash = 89 * hash + this.numberOfBooks;
        return hash;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (this == obj)
        {
            return true;
        }
        if (obj == null)
        {
            return false;
        }
        if (getClass() != obj.getClass())
        {
            return false;
        }
        final TDatosRepositorio other = (TDatosRepositorio) obj;
        if (this.numberOfBooks != other.numberOfBooks)
        {
            return false;
        }
        if (!Objects.equals(this.repositoryName, other.repositoryName))
        {
            return false;
        }
        return Objects.equals(this.repositoryAddress, other.repositoryAddress);
    }
    
}

File: Practica 3 - C# Net Remoting/GestorBibliotecaService/GestorBibliotecaService/TDatosRepositorio.cs
using GestorBibliotecaService.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GestorBibliotecaService
{
    public class TDatosRepositorio : IEquatable<TDatosRepositorio>
    {
        private string repositoryName;
        private string repositoryAddress;
        private int numberOfBooks;
        private string repositoryFilePath;

        private BookRepository bookRepository;

        public TDatosRepositorio()
        {
        }

        public TDatosRepositorio(string repositoryName, string repositoryAddress, int numberOfBooks,
                BookRepository bookRepository, string repositoryFilePath)
        {
            this.repositoryName = repositoryName;
            this.repositoryAddress = repositoryAddress;
            this.numberOfBooks = numberOfBooks;
            this.bookRepository = bookRepository;
            this.repositoryFilePath = repositoryFilePath;
        }

        public string RepositoryName
        {
            get { return this.repositoryName; }
            set { this.repositoryName = value; }
        }

        public string RepositoryAddress
        {
            get { return this.repositoryAddress; }
            set { this.repositoryAddress = value; }
        }

        public int NumberOfBooks
        {
            get { return this.numberOfBooks; }
            set { this.numberOfBooks = value; }
        }

        public BookRepository BookRepository
        {
            get { return this.bookRepository; }
            set { this.bookRepository = value; }
        }

        public string RepositoryFilePath
        {
            get { return this.repositoryFilePath; }
            set { this.repositoryFilePath = value; }
        }

        public override bool Equals(object obj)
        {
            return Equals(obj as TDatosRepositorio);
        }

        public bool Equals(TDatosRepositorio other)
        {
            return other != null &&
                   repositoryName == other.repositoryName &&
                   repositoryAddress == other.repositoryAddress;
        }

        public override int GetHashCode()
        {
            int hashCode = -1138466103;
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(repositoryName);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(repositoryAddress);
            return hashCode;
        }
    }
}
